core/src/main/java/io/grpc/internal/Stream.java

```diff
@@ -77,4 +79,10 @@
    * result in excessive buffering within the transport.
    */
   boolean isReady();
+
+  /**
+   * Sets the default message encoder for messages on this stream.
+   * @param c the compressor / decompressor
+   */
+  void setDefaultCodec(Codec c);
```

Comment:
Now that you ask.... I suppose not.  The spec says that you should be able to request that the whole channel use a compressor, ask that an individual call use a compressor different than the channel default, and also to be able to selectively turn off compression for each message.  

Currently the presence of a Codec implies that compression is desired even those these two ideas are separable.  Because this feature is being added in chunks, I personally expected this API to change slightly with each commit until it is ready to be released.   (it is easier to see problems once the entire thing is in place, rather than at this early stage)